### DISCUSSION TITLE
Error 503 for pages with special file name characters

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#491] [Report a bug] [answered]
There seems to be a problem in the search extension if you have content files with special characters in the file name. 

```
---
Title: Zurücksetzen des Digitalradios Dual DAB4
Layout: wiki
Tag: Anleitung, Radio
---
Page text.
```

This will create an URL like `https://website/wiki/zurücksetzen-des-digitalradios-dual-dab4` upon page creation from the editor. Now if I use the search extension, the server responds with an error 503 after entering a query. A workaround is to use the `TitleSlug` setting or manually rename the appropriate markdown file. The strange thing is that I only can reproduce this error on a system where php 7.3 runs as FPM. The search extension works fine on another machine with the same PHP version running as apache module. Therefore I'm not sure if this is a general error. 

Get in touch if you need a testing playground. 😀

Steffen

**Comment by markseuffert** [comment]
Can you check if there's more information in the Apache and PHP-FPM logs?

**Comment by schulle4u** [comment]
The apache error log prints the following two lines for each failed search query (slightly edited the referer): 

```
[Sun Feb 09 11:03:13.763834 2020] [proxy_fcgi:error] [pid 7153:tid 140064128599808] [client 91.57.18.20:50180] AH01067: Failed to read FastCGI header, referer: https://subdomain.website/search/
[Sun Feb 09 11:03:13.763851 2020] [proxy_fcgi:error] [pid 7153:tid 140064128599808] (104)Connection reset by peer: [client 91.57.18.20:50180] AH01075: Error dispatching request to : , referer: https://subdomain.website/search/
```

PHP7.3-fpm log:
```
[09-Feb-2020 11:03:13] WARNING: [pool powerradio4u] child 5981 exited on signal 11 (SIGSEGV) after 8.419564 seconds from start
```

**Comment by markseuffert** [comment]
A SIGSEGV points to a software crash during execution. It could be a bug in PHP-FPM.

**Comment by schulle4u** [comment]
My hosting panel allows me to change the used php version, so I did a test using php 7.4.0. This results in the following php error: 

`[Sun Feb 09 12:02:40.754331 2020] [proxy_fcgi:error] [pid 8111:tid 140064388257536] [client 91.57.18.20:50477] AH01071: Got error 'PHP message: PHP Fatal error:  Allowed memory size of 537919488 bytes exhausted (tried to allocate 262144 bytes) in /home/[...]/website/system/extensions/core.php on line 941', referer: https://subdomain.website/search/`

Maybe I can't calculate anymore, but actually 262144 bytes shouldn't exhaust the available memory? 😖

**Comment by schulle4u** [comment]
Problem solved, it was caused by a non-utf8 ftp connection while transfering files.

**Comment by markseuffert** [comment, marked as answer]
This was a bug. A file name with invalid UTF8 character caused `preg_replace()` to return an empty string, which then caused an endless recursive function call and after some time the PHP process consumed all available memory. The latest core checks for invalid UTF-8 characters.

Thanks for providing a testing playground. Should be fixed now.
